src/Tickets/QR/Connector.php

```diff
@@ -0,0 +1,237 @@
+	 * @param string     $security_code
+	 *
+	 * @return string
+	 */
+	public function get_checkin_url( $ticket_id, $event_id, string $security_code ): string {
+		$base_url = home_url( '/' );
+
+		/**
+		 * Allows filtering the base URL which QR code query args are appended to. Defaults to
+		 * the site's home_url() with a trailing slash.
+		 *
+		 * @since      4.7.3
+		 * @deprecated TBD Use `tec_tickets_qr_code_base_url` instead.
+		 *
+		 * @param string     $base_url
+		 * @param int|string $ticket_id
+		 * @param int|string $event_id
+		 *
+		 */
+		$base_url = apply_filters_deprecated( 'tribe_tickets_qr_code_base_url', [ $base_url, $ticket_id, $event_id ], 'TBD', 'Use `tec_tickets_qr_code_base_url` instead.' );
+		/**
+		 * Allows filtering the base URL which QR code query args are appended to. Defaults to
+		 * the site's home_url() with a trailing slash.
+		 *
+		 * @since TBD
+		 *
+		 * @param string     $base_url      The base URL.
+		 * @param int|string $ticket_id     The ticket ID.
+		 * @param int|string $event_id      The event ID.
+		 * @param string     $security_code The security code from the attendee.
+		 */
+		$base_url = apply_filters( 'tec_tickets_qr_code_base_url', $base_url, $ticket_id, $event_id, $security_code );
+
+		$query_args = [
+			'event_qr_code' => 1,
+			'ticket_id'     => 1,
+			'event_id'      => $event_id,
+			'security_code' => $security_code,
+			'path'          => urlencode( tribe_tickets_rest_url_prefix() . '/qr' ),
+		];
+
+		$url = add_query_arg( $query_args, $base_url );
+
+		/**
+		 * Allows filtering the checkin url.
+		 *
+		 * @since TBD
+		 *
+		 * @param string     $url
+		 * @param int|string $ticket_id
+		 * @param int|string $event_id
+		 * @param string     $security_code
+		 *
+		 */
+		return apply_filters( 'tec_tickets_qr_code_checkin_url', $url, $ticket_id, $event_id, $security_code );
+	}
+
+	/**
+	 * Get QR Code URL from ticket.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $ticket Ticket data we are using to get the QR code image from.
+	 *
+	 * @return ?string
+	 */
+	public function get_image_url_from_ticket_data( $ticket ): ?string {
+		if ( tribe( Settings::class )->is_enabled( $ticket ) ) {
+			return null;
+		}
+
+		$link = $this->get_checkin_url( $ticket['qr_ticket_id'], $ticket['event_id'], $ticket['security_code'] );
+		$qr   = $this->get_image_url_for_link( $link );
+
+		if ( ! $qr ) {
+			return null;
+		}
+
+		return $qr;
+	}
+
+	/**
+	 * Generates the QR image for a given link and stores it in /wp-content/uploads.
+	 * Returns the link to the new image.
+	 *
+	 * @param ?string $link
+	 *
+	 * @return ?string
+	 */
+	public function get_image_url_for_link( ?string $link ): ?string {
+		$qr_code = tribe( QR::class );
+
+		if ( empty( $link ) ) {
+			return null;
+		}
+
+		if ( is_wp_error( $qr_code ) ) {
+			return null;
+		}
+
+		// @ToDo @rafsuntaskin @juanfra change the default directory to `tec/tickets-plus/qr-codes` for storing images.
```

Comment:
Is this @todo relevant anymore since QRs will be accessible from different plugins?

Reply:
Partially so I will modify.